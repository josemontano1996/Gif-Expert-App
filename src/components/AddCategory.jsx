import { useState } from 'react';

import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState('');

  const onInputChage = (event) => {
    setInputValue(event.target.value);
  };

  const onSubmit = (event) => {

    event.preventDefault();
    const newInputValue = inputValue.trim();
    if (newInputValue.length <= 1) return;

    /*   setCategories((categories) => [inputValue, ...categories]); */
    onNewCategory(newInputValue);

    setInputValue('');
  };

  return (
    <form onSubmit={onSubmit} aria-label='form'>
      <input
        type='text'
        placeholder='Search for Gifs'
        value={inputValue}
        onChange={onInputChage}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
