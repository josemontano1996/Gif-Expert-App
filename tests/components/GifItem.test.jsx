import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('<GifItem /> testing', () => {
  const title = 'One Piece';
  const url = `https://one-piece.com/onepunch.jpg`;

  test('Evaluate Snapshot`', () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test('Should show img with the given URL and alt', () => {
    render(<GifItem title={title} url={url} />);
    /* screen.debug(); */

    const { src, alt } = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test('Must show component title', () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
