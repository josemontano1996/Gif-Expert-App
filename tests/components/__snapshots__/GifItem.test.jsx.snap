// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem testing Evaluate Snapshot\` 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One Piece"
      src="https://one-piece.com/onepunch.jpg"
    />
    <p>
      One Piece
    </p>
  </div>
</div>
`;
